packages/firestore/src/core/sync_engine.ts

```diff
@@ -1100,6 +1100,11 @@ class MultiTabSyncEngineImpl extends SyncEngineImpl {
     }
   }
 
+  async synchronizeWithChangedDocuments(): Promise<void> {
+    const changes = await this.localStore.getNewDocumentChanges();
+    await this.emitNewSnapsAndNotifyLocalStore(changes);
+  }
```

Comment:
This is the style of method where I would advocate to drop the await keyword and simply return the Promise. See here for the code overhead of using `await` in ES5: https://www.typescriptlang.org/play/?target=1#code/IYZwngdgxgBAZgV2gFwJYHsL3egFAShgG8AoGcmYAd2FWRggFMqYAFAJ3QFtURHcCMALwA+YgF98AbhLigA

Note that __awaiter and __generator are already part of our code base, so we are only looking at the overhead in `foo`.